videoframegrid.py

```diff
@@ -0,0 +1,130 @@
+# -*- coding: utf-8 -*-
+
+import sip
+
+from PyQt5 import QtWidgets
+from coordinates import VideoFrameCoordinates
+from videoframes import LiveVideoFrame
+from config import cfg
+from constants import CONFIG_MUTE
+
+
+class VideoFrameGrid(QtWidgets.QGridLayout):
+    """The VideoFrameGrid is the GridLayout container for all LiveVideoFrames.
+
+    The VideoFrameGrid keeps track of the position of all LiveVideoFrames and
+    is in charge of reparenting a VideoFrame back to it's original position
+    after a toggled fullscreen mode.
+    """
+
+    def __init__(self):
+        """Creates a new VideoFrameGrid.
+
+        The VidoeFrameGrid holds a list of the active videoframe objects as
+        well as a coordinates object to keep track of the current grid
+        coordinates (beginning at (0, 0)).
+        """
+        super(VideoFrameGrid, self).__init__()
+        self.videoframes = []
+        self.coordinates = VideoFrameCoordinates(x=0, y=0)
+        self.selected_frame = None
+
+    def _add_videoframe(self, videoframe):
+        """Adds the provided videoframeobject to the VideoFrameGrid."""
+        self.videoframes.append(videoframe)
+        self.addWidget(videoframe, self.coordinates.x, self.coordinates.y)
+        self.coordinates = self.coordinates.update_coordinates()
+
+    def _create_videoframe(self, vlc_instance, stream_options, quality):
+        """Creates a new LiveVideoFrame object."""
+        return LiveVideoFrame(vlc_instance, stream_options, quality)
+
+    def add_new_videoframe(self, vlc_instance, stream_options, quality):
+        """Creates and adds a new LiveVideoFrame to the VideoFrameGrid."""
+        videoframe = self._create_videoframe(vlc_instance, stream_options, quality)
+        videoframe.player.audio_set_mute(cfg[CONFIG_MUTE])
+        videoframe._swap = self.swap_frame
+        videoframe._coordinates = self.coordinates
+        videoframe._delete_stream = self.delete_stream
+        self._add_videoframe(videoframe)
+
+    def swap_frame(self, frame):
```

Comment:
I've only moved the functions from `main` but I think we should refactor the following lines (51 and forward) a bit, but I'll leave that for a separate PR.